lib/modules/datasource/go/base.ts

```diff
@@ -154,17 +155,35 @@ export class BaseGoDatasource {
       const parsedUrl = URL.parse(goSourceUrl);
 
       // TODO: `parsedUrl.pathname` can be undefined
-      const packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
+      let packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
 
-      const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
+      const endpoint = GlobalConfig.get('endpoint');
```

Comment:
```suggestion
      const endpoint = GlobalConfig.get('endpoint')!;
```
you can assume that `endpoint` isn't undefined here.